qcodes/instrument/channel.py

```diff
@@ -167,9 +140,10 @@ class ChannelList(Metadatable):
 
     """
 
-    def __init__(self, parent, name, chan_type: type, chan_list=None,
+    def __init__(self, parent: Instrument, name, chan_type: type, chan_list=None,
```

Comment:
Why not, it's gradual typing after all ;) That aside it's easy to add them but the main benefit comes from annotating the `chan_type` because the linter has trouble figuring out that we are passing a Class and not an instance of a Class but sure it's easy to add the rest. 

I am not sure that there is a better way of annotating that this is a Class not an instance of the Class. type(multichan_paramclass) is not what you are looking for at least.